[Javier] ip

Javiier-pzk opened the pull request:
# DukePro

> Duke is goooooooood -Anonymous

DukePro frees your mind of having to remember things you need to do. It's,

* Text-based
* Easy to learn
* ~~FAST~~ *SUPER* FAST to use
* Responsive

All you need to do is,

  1. Download it from [here](https://github.com/Javiier-pzk/ip/releases/tag/v1.0)
  2. Double-click it.
  3. Add your tasks.
  4. Let it manage your tasks for you 😎 

And its **FREE**!

Features:

- [x] Managing to-dos
- [x] Managing deadlines
- [x] Managing events
- [x] Search for tasks
- [x] Marks tasks as done or delete tasks!
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the super complex `main` method:

```java
public static void main(String[] args) {
    new Duke(FILEPATH).run();
}
```

Commit:
Add Gradle support

Javiier-pzk committed:
Ignore .class files

Javiier-pzk committed:
Implement Greet, Echo, Exit

Javiier-pzk committed:
Level 1 TextUiTest

Javiier-pzk committed:
Implement Add, List

Javiier-pzk committed:
Level 2 TextUiTest

Javiier-pzk committed:
Add Task class

Javiier-pzk committed:
Implement Mark as Done

Javiier-pzk committed:
Level 3 TextUiTest

Javiier-pzk committed:
Implement ToDo, Event and Deadline classes and add toString() method to Task class

Javiier-pzk committed:
Add support for tracking ToDos, Events and Deadlines

Javiier-pzk committed:
Level 4 TextUiTest

Javiier-pzk committed:
Add getter

Javiier-pzk committed:
Implement error handling

Javiier-pzk committed:
Rewrite code structure for Duke.java

Javiier-pzk committed:
Level 5 TextUiTest

Javiier-pzk committed:
Implement delete function

Javiier-pzk committed:
Level 6 TextUiTest

Javiier-pzk committed:
Package commands into individual classes

Javiier-pzk committed:
Add save to file format

Javiier-pzk committed:
Implement reading and writing to files.

Javiier-pzk committed:
Create new Exception classes for Duke

Javiier-pzk committed:
Implement Date and Times

Javiier-pzk committed:
Level-8-TextUitest

Javiier-pzk committed:
Merge commit '9e386d9c0be96abba65a9628c166a25cb8079a92'

* commit '9e386d9c0be96abba65a9628c166a25cb8079a92':
  Add save to file format

Javiier-pzk committed:
Merge branch 'branch-Level-7'

Javiier-pzk committed:
Merge commit '869a2b7f8d9a4e18f50d41ecf42081a988cc1a84'

Javiier-pzk committed:
Merge commit '0c8b681e92a4d35d13d952e6979ed5568dc69ba7'

* commit '0c8b681e92a4d35d13d952e6979ed5568dc69ba7':
  Implement Date and Times

# Conflicts: (Resolved)
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Javiier-pzk committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level-8-TextUitest

Javiier-pzk committed:
Implement Parser Storage, TaskList and Ui classes

Javiier-pzk committed:
Implement more-oop into existing classes

Javiier-pzk committed:
Add minor error message changes

Javiier-pzk committed:
Divide classes into packages

Javiier-pzk committed:
JUnit test for Deadline and Task class

Javiier-pzk committed:
Implement search for task

Javiier-pzk committed:
Add JavaDocs to all classes

Javiier-pzk committed:
Fix bugs from Event and Deadline class

Javiier-pzk committed:
Fix test cases

Javiier-pzk committed:
Merge branch 'master' into branch-A-JavaDoc

master:
Fix test cases
Fix bugs from Event and Deadline class

Javiier-pzk committed:
Merge branch 'master' into branch-Level-9

* master:
  Fix test cases
  Fix bugs from Event and Deadline class

Javiier-pzk committed:
Adjust code to comply with coding standard

Javiier-pzk committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Adjust code to comply with coding standard

Javiier-pzk committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Implement search for task

Javiier-pzk committed:
Fix merge conflict

Javiier-pzk committed:
Adjust code to comply with coding standards

Javiier-pzk committed:
Add JavaDocs to respective classes.

Javiier-pzk committed:
Change saving directory from Desktop to data

Javiier-pzk committed:
Ignore data directory

Javiier-pzk committed:
Fix coding standard violations

ooawagaeri commented on src/Test/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,115 @@
+import duke.exception.InvalidCommandException;
+import org.junit.jupiter.api.Test;
+import java.time.DateTimeException;
+import static org.junit.jupiter.api.Assertions.*;
```
Order of import should be: static, normal, other packages

ooawagaeri commented on src/Test/duke/task/TaskTest.java:
```diff
@@ -0,0 +1,46 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Import should always be listed explicit

ooawagaeri replied:
Order of import should be: static, normal, other packages

ooawagaeri commented on src/Test/duke/task/TaskTest.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Test
+    void testToString() {
```
Why are your testing methods different from DeadlineTest?

ooawagaeri commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,43 @@
+package duke;
+
+import duke.command.*;
```
Imported classes should always be listed explicitly

ooawagaeri commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,107 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
```
Order of import should be: normal, other packages

ooawagaeri commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,134 @@
+package duke;
+
+import duke.task.Task;
+import java.util.ArrayList;
```
Order of import should be: static, normal, other packages

ooawagaeri commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,71 @@
+import duke.task.Deadline;
+import duke.task.Task;
+import java.io.IOException;
+import java.time.DateTimeException;
```
Order of import should be: normal, other packages

ooawagaeri commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,53 @@
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import java.io.IOException;
```
Order of import should be: normal, other packages

ooawagaeri commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,18 @@
+        return true;
+    }
+
+    @Override
```
Missing method comment, should be consistent with other classes

ooawagaeri commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,71 @@
+
+    /**
+     * Executes a "deadline" command.
+     *
```
Perhaps you can add more detail? A bit more description?

ooawagaeri commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,53 @@
+
+    /**
+     * Executes a "delete" command.
+     *
```
Perhaps you can add more detail? A bit more description?

ooawagaeri commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,54 @@
+
+    /**
+     * Executes a "done" command.
+     *
```
Perhaps you can add more detail? A bit more description?

ooawagaeri commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,54 @@
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import java.io.IOException;
```
Order of import should be: normal, other packages

ooawagaeri commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,71 @@
+import duke.task.Event;
+import duke.task.Task;
+import java.io.IOException;
+import java.time.DateTimeException;
```
Order of import should be: normal, other packages

ooawagaeri commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,71 @@
+
+    /**
+     * Executes an "event" command.
+     *
```
Perhaps you can add more detail? A bit more description?

ooawagaeri commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,71 @@
+                taskList.add(event);
+                storage.save(taskList);
+            } catch (DateTimeException exception) {
+                ui.showError(
```
Maybe can have a specific showDateError?

ooawagaeri commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,44 @@
+
+    /**
+     * Executes a "find" command.
+     *
```
Perhaps you can add more detail? A bit more description?

ooawagaeri commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+    /**
+     * Executes a "list" command.
+     *
```
Perhaps you can add more detail? A bit more description?

ooawagaeri commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,53 @@
+
+    /**
+     * Executes a "todo" command.
+     *
```
Perhaps you can add more detail? A bit more description?

ooawagaeri commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,53 @@
+import duke.Ui;
+import duke.task.Task;
+import duke.task.ToDo;
+import java.io.IOException;
```
Order of import should be: normal, other packages

ooawagaeri commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,99 @@
+import duke.exception.InvalidCommandException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
```
Order of import should be: normal, other packages

ooawagaeri commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,100 @@
+import duke.exception.InvalidCommandException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
```
Order of import should be: normal, other packages

ooawagaeri reviewed (commented):
Good day Javier,

Honestly a good job for majority of your coding standard. 
The comments made mostly consist of incorrect importing and some commenting suggestions. Don't take anything personally :) Just trying to help.

Good luck for the rest of you iP.

Javiier-pzk committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Add Gradle support

Javiier-pzk committed:
Change mainClassName in build.gradle

Javiier-pzk committed:
Reorganise classes into neater packages

Javiier-pzk committed:
Reorganise test classes into neater packages

Javiier-pzk committed:
Add CheckStyle config files

Javiier-pzk committed:
Update CheckStyle version

Javiier-pzk committed:
Fix coding standards violations

Javiier-pzk committed:
Merge branch 'branch-Checkstyle'

* branch-Checkstyle:
  Fix coding standards violations
  Update CheckStyle version
  Add CheckStyle config files

Javiier-pzk committed:
Fix coding standard violations for test classes

Javiier-pzk committed:
Merge branch 'branch-Checkstyle'

* branch-Checkstyle:
  Fix coding standard violations for test classes

Javiier-pzk committed:
Implement basic GUI using JavaFX

Javiier-pzk committed:
Add user and duke chat avatars

Javiier-pzk committed:
Add dialog box and image view functionalities

Javiier-pzk committed:
Add Launcher and DialogBox classes into GUI package

Javiier-pzk committed:
Reorganise Duke class

Javiier-pzk committed:
Add FXML files for MainWindow and DialogBox

Javiier-pzk committed:
Implement Duke GUI using FXML

Javiier-pzk committed:
Add constructors and change return type of execute() method

Javiier-pzk committed:
Change methods to return strings instead of void

Javiier-pzk committed:
Change Command class constructors

Javiier-pzk committed:
Replace run() method with getResponse()

Javiier-pzk committed:
Update JavaDocs for existing and new classes

Javiier-pzk committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Update JavaDocs for existing and new classes
  Replace run() method with getResponse()
  Change Command class constructors
  Change methods to return strings instead of void
  Add constructors and change return type of execute() method
  Implement Duke GUI using FXML
  Add FXML files for MainWindow and DialogBox
  Reorganise Duke class
  Add Launcher and DialogBox classes into GUI package
  Add dialog box and image view functionalities
  Add user and duke chat avatars
  Implement basic GUI using JavaFX

Javiier-pzk committed:
Add missing JavaDocs

Javiier-pzk committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add missing JavaDocs

Javiier-pzk committed:
Update JavaDocs description and test method naming

Javiier-pzk committed:
Update test methods

Javiier-pzk committed:
Change method names to fit naming convention

timiditi commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+     * @param index The index of the task to mark as completed in the taskList.
+     * @param storage An instance of a the Storage class that saves and loads Duke's data.
+     * @return A string representing the task that has been marked as completed.
+     */
```
Do you want to document the exception this method `@throws`?

timiditi reviewed (commented):
LGTM! Very nice job with the code.

Javiier-pzk committed:
Adjust margins and change dialog images

Javiier-pzk committed:
Restructure code to improve readability and code quality.

Current code in the Command classes have many duplicate error strings which can be reduced by
putting them in the Ui class. The Command classes and TaskList class also do not follow the SLAP principle.

The Command and TaskList classes have been further abstracted such that all error messages are in the Ui class.
The Ui class now contains all the strings which Duke will return when executing a command or returning error messages.
This is so anything to do with Duke's replies to the user can be found all in one place and it is easier to fix bugs if any.
The code is also now easier to read and much cleaner.

Javiier-pzk committed:
Restructure code in Task classes using guard clauses

Javiier-pzk committed:
Change variable names to be less confusing in the Parser class.

Javiier-pzk committed:
Add getBy() and getAt() method to Deadline and Event classes respectively

Javiier-pzk committed:
Add assertions to verify assumptions in Command subclass methods

Javiier-pzk committed:
Merge pull request #1 from Javiier-pzk/branch-A-Assertions

Add assertions to verify assumptions made in some methods.

Javiier-pzk committed:
Merge branch 'master' into branch-A-CodeQuality

Javiier-pzk committed:
Merge pull request #2 from Javiier-pzk/branch-A-CodeQuality

Restructure code to improve code quality and readability.

Javiier-pzk committed:
Merge branch 'master' of https://github.com/Javiier-pzk/ip

* 'master' of https://github.com/Javiier-pzk/ip:
  Add assertions to verify assumptions in Command subclass methods
  Add getBy() and getAt() method to Deadline and Event classes respectively
  Change variable names to be less confusing in the Parser class.
  Restructure code in Task classes using guard clauses
  Restructure code to improve readability and code quality.

Javiier-pzk committed:
Minor bug fix from merging pull requests

Javiier-pzk committed:
Merge branch 'master' into branch-A-Assertions

* master:
  Minor bug fix from merging pull requests
  Change variable names to be less confusing in the Parser class.
  Restructure code in Task classes using guard clauses
  Restructure code to improve readability and code quality.

Javiier-pzk committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Minor bug fix from merging pull requests
  Add assertions to verify assumptions in Command subclass methods
  Add getBy() and getAt() method to Deadline and Event classes respectively

Javiier-pzk committed:
Edit test cases in DeadlineTest

Javiier-pzk committed:
Set up and add JUnit tests to GitHub Actions CI

Javiier-pzk committed:
Add ability for Duke to detect duplicate tasks when adding tasks

Duke cannot detect duplicate tasks previously, allowing the user to add multiple of
the same tasks. This makes it more user friendly as the user might not remember if
he or she has added a specific task before and thus add it multiple times which can
cause confusion.

Let's override the equals method and define our own equals method in the 3 Task subclasses.

Polymorphism allows us to override the equals method in each of the 3 Task subclasses
to compare the 3 types of tasks differently.  Hence, we can check for any equal task in the
taskList before adding a new task in with just one loop

Javiier-pzk committed:
Add generateTaskString() method to reduce duplicated code

Javiier-pzk committed:
Merge branch 'branch-C-DetectDuplicates'

* branch-C-DetectDuplicates:
  Add generateTaskString() method to reduce duplicated code
  Add ability for Duke to detect duplicate tasks when adding tasks

Javiier-pzk committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:

Javiier-pzk committed:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:

Javiier-pzk committed:
Make search method more flexible
Search is able to search task the whole task string instead of just task description

Javiier-pzk committed:
Update README.md

Add user guide

Javiier-pzk committed:
Add Ui screenshot

Javiier-pzk committed:
Merge branch 'master' of https://github.com/Javiier-pzk/ip

* 'master' of https://github.com/Javiier-pzk/ip:
  Update README.md

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Update Javadocs

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Fix font issue problem with MacOS

Javiier-pzk committed:
Fix font issue on MacOS

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Create README.md

Javiier-pzk committed:
Update README.md

Javiier-pzk committed:
Add exit from GUI if bye command is executed

Javiier-pzk committed:
Update README.md
